### PR TITLE
added university page and filter

### PR DESCRIPTION
**Pull request by MxolisiSibaya322**

### CODE REVIEW

File: JS/dashboard.js [hunk 1]
```diff
@@ -64,6 +64,11 @@ function loadScriptsBySection(sectionName) {
     case "StudentRequest":
```
Comment (Madimetja94) [inline JS/dashboard.js:64]: Changing thw vars to let or consts.

File: JS/studentRequest.js [hunk 2]
```diff
@@ -53,145 +50,134 @@ async function GetAllRequests(){
...
+    fetch("http://localhost:5263/api/StudentFundRequest/create", options)
+      .then((respons) => respons.json())
+      .then((data) => window.alert(data.message));
+  });
 
 function getUniqueColumnValues(table, columnIndex) {
-    var values = [];
-    var rows = table.getElementsByTagName("tr");
+  let values = [];
+  let rows = table.getElementsByTagName("tr");
```
Comment (Madimetja94) [inline JS/studentRequest.js:117]: change variables that are not re-assigned different values to const.

File: JS/AllUniversities.js [hunk 3]
```diff
@@ -1,18 +1,136 @@
...
+    bodyTemp: bodyMessage,
+  };
+}
+async function getAllUniversity() {
+  const response = await fetch(
+    "http://localhost:5263/Admin/GetAllUniversityRequests",
+    createOptions("GET", {})
+  );
+  const dataResponse = await response.json();
+  console.log(dataResponse);
```
Comment (Madimetja94) [inline JS/AllUniversities.js:19]: Remove the console logs after testing.

File: JS/AllUniversities.js [hunk 4]
```diff
@@ -1,18 +1,136 @@
...
+    const th = document.createElement("th");
+    th.textContent = key;
+    headerRow.appendChild(th);
+  });
+
+  table.appendChild(headerRow);
 
-        },
-        bodyTemp: bodyMessage,
+  dataResponse.forEach((obj) => {
```
Comment (Madimetja94) [inline JS/AllUniversities.js:44]: use full namings , obj => object

File: JS/AllUniversities.js [hunk 5]
```diff
@@ -1,18 +1,136 @@
...
+      row.appendChild(cell);
     });
-    
+    table.appendChild(row);
+  });
+  display.appendChild(table);
+  populateFilterOptions();
+}
+function getUniqueColumnValues(table, columnIndex) {
+  let values = [];
```
Comment (Madimetja94) [inline JS/AllUniversities.js:68]: declare with const. since the array wont be re-assinged another address.

File: JS/AllUniversities.js [hunk 6]
```diff
@@ -1,18 +1,136 @@
...
+  return {
+    method: methodName,
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${sessionStorage.getItem("token")}`,
+    },
+    bodyTemp: bodyMessage,
+  };
+}
+async function getAllUniversity() {
```
Comment (Madimetja94) [inline JS/AllUniversities.js:13]: rename to getAllUniversityRequests to make descriptive.

**Review by Keith-Hughes — APPROVED** [review]

File: JS/studentRequest.js [hunk 7]
```diff
@@ -53,145 +50,134 @@ async function GetAllRequests(){
...
+    document.getElementById("StudentRequestTable"),
+    5
+  ); // Assuming status is in the sixth column
+
+  populateDropdown(universityFilter, universityValues);
+  populateDropdown(statusFilter, statusValues);
 }
 
 // Helper function to populate a dropdown with options
 function populateDropdown(selectElement, values) {
```
Comment (Madimetja94) [inline JS/studentRequest.js:176]: I was thinking if possible we find a way to store helper functions in on file then we re-use then wherever we need them.

**Review by Madimetja94 — CHANGES_REQUESTED** [review]

**Review by Madimetja94 — COMMENTED** [review]
Great work Mxolisi, everything works so far. Just a few small changes then we are good.

**Pull request merged**